GitCommands/Remotes/GitRemoteManager.cs

```diff
@@ -127,15 +127,50 @@ bool IsSettingForBranch(string setting, string branchName)
         }
 
         /// <summary>
-        /// Retrieves disabled remotes from the .git/config file.
+        /// Retrieves disabled remotes from .git/config file
         /// </summary>
-        public string[] GetDisabledRemotes()
+        public IReadOnlyList<Remote> GetDisabledRemotes()
+        {
+            var disabledRemotes = GetModule().LocalConfigFile.GetConfigSections()
+                .Where(s => s.SectionName == $"{DisabledSectionPrefix}remote")
+                .Select(s => new Remote(s.SubSection, s.GetValue("pushurl", s.GetValue("url")), s.GetValue("url")))
+                .ToList();
+
+            return disabledRemotes;
+        }
+
+        /// <summary>
+        /// Retrieves disabled remote names from the .git/config file.
+        /// </summary>
+        public IReadOnlyList<string> GetDisabledRemoteNames()
         {
             var module = GetModule();
             return module.LocalConfigFile.GetConfigSections()
-                                         .Where(s => s.SectionName == $"{DisabledSectionPrefix}remote")
-                                         .Select(s => s.SubSection)
-                                         .ToArray();
+                .Where(s => s.SectionName == $"{DisabledSectionPrefix}remote")
+                .Select(s => s.SubSection)
+                .ToList();
+        }
+
+        /// <summary>
+        /// Retrieves enabled remote names
+        /// </summary>
+        public IReadOnlyList<string> GetEnabledRemoteNames()
+        {
+            return GetModule().GetRemoteNames();
+        }
+
+        /// <summary>
+        /// Retrieves enabled remote names of remotes without branches (i.e. that require a fetch)
+        /// </summary>
+        public IReadOnlyList<string> GetEnabledRemotesNameWithoutBranches()
+        {
+            var branches = GetModule().GetRefs()
+                .Where(branch => branch.IsRemote && !branch.IsTag)
+                .Select(branch => branch.Name);
+
+            return GetEnabledRemoteNames()
+                .Where(remote => branches.FirstOrDefault(branch => branch.StartsWith(remote)) == default)
```

Comment:
💭 Just to confirm, we are not doing multiple enumerations of `branches` here, are we?

Reply:
We are, actually. This isn't very efficient. I'll rewrite to be more efficient.

Reply:
Okay, so the code was originally this:

```csharp
            var branches = GetModule().GetRefs()
                .Where(branch => branch.IsRemote && !branch.IsTag)
                .Select(branch => branch.Name);

            return GetEnabledRemoteNames()
                .Where(remote => branches.FirstOrDefault(branch => branch.StartsWith(remote)) == default)
                .ToList();
```

Which, as you realized, was not very efficient as it would iterate over all branches for each enabled remote name, seeing if the remote does not appear in any of the branch names.

I rewrote it to, instead, build a set of remotes that have branches, and then subtracting it from the set of all enabled remotes:

```csharp
            HashSet<string> remotesWithBranches = GetModule().GetRefs()
                .Where(branch => branch.IsRemote && !branch.IsTag)
                .Select(branch => branch.Name.SubstringUntil('/'))
                .ToHashSet();

            HashSet<string> allRemotes = GetEnabledRemoteNames().ToHashSet();

            return allRemotes.Except(remotesWithBranches).ToList();
```

This scales better, though for a small set of remotes and branches, this probably performs worse. Still, scalability is more important. Is this good for you, @RussKie?

Reply:
That's even better than I thought 💯